File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let memo =[];
    s = [...s];
    
    let answer = s.map((e, i)=>{
        let result = -1;
        
        if(i == 0){
            memo.push(e);
            return result;
        }  
        
        for(let x = 0; x < memo.length; x++){
            if(memo[x] === e){
                result = i-x;
            }
        }
        
        memo.push(e);
        return result;
    })

    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    stack = []
    for ch in s:
        if(len(stack)==0):
            answer.append(-1)
        else:
            result = -1
            for i,target in enumerate(reversed(stack)):
                if(target == ch):
                    result = i + 1
                    break
            answer.append(result)   
        stack.append(ch)
        
    return answer